tee-worker/litentry/core/assertion-build/src/achainable/amount.rs

```diff
@@ -103,13 +108,30 @@ pub fn build_amount(req: &AssertionBuildRequest, param: AchainableAmount) -> Res
 		.collect::<Vec<String>>();
 
 	let achainable_param = AchainableParams::Amount(param.clone());
-	let flag = request_achainable(addresses, achainable_param.clone())?;
+	let bname = AchainableNameBalance::from(achainable_param.name()).map_err(|e| {
+		Error::RequestVCFailed(
+			Assertion::Achainable(achainable_param.clone()),
+			e.into_error_detail(),
+		)
+	})?;
+	let mut balance = "".to_string();
+	let mut flag = false;
+	if bname == AchainableNameBalance::BalanceUnderAmount {
+		balance = request_achainable_balance(addresses, achainable_param.clone())?;
```

Comment:
I can't really tell from the names what's the difference between `request_achainable_balance` and `request_achainable` - their distinct return types make it even more mysterious.

Maybe they are either sloppily named or implemented :(

Reply:
`request_achainable` is the standard achainable API interface, which only requires the API for `true/false` results, and later added `_balance` and others require special handling of the results returned by achainable, such as parsing balance from the `text` field. Perhaps the naming of this is too simple and crude.